File: 프로그래머스/lv2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    sort(people.begin(), people.end());
    int i = 0, back;
    
    while(people.size() > i) {
        int back = people.back();
        people.pop_back();
        if(people[i] + back <= limit) {
            answer++; i++;
        }
        else answer++;
    }
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    start, end = 0, len(people) - 1
    
    while start <= end:
        flag = True
        if people[start] + people[end] <= limit:
            start += 1
            end -= 1
        else:
            end -= 1
        answer += 1        
    
    return answer